File: PrefixSum/java/FindPivotIndex.java
package PrefixSum.java;

import java.util.Arrays;

public class FindPivotIndex {
    public int pivotIndex(int[] nums) {
        int total = Arrays.stream(nums).sum();
        int left = 0;

        for (int i = 0; i < nums.length; i++) {
            int right = total - left - nums[i];

            if (right == left) {
                return i;
            }
            left += nums[i];
        }

        return -1;
    }

}

File: PrefixSum/FindPivotIndex.scala
object Solution {
  def pivotIndex(nums: Array[Int]): Int = {
    var l = 0
    var r = nums.sum
    var ind = -1
    var found = false
    for (i <- nums.indices) {
      if (i > 0) l += nums(i - 1)
      r -= nums(i)
      if (r == l && !found) {
        ind = i
        found = true
      }
    }
    ind

  }
}
